lib/galaxy/tool_shed/galaxy_install/tools/tool_panel_manager.py

```diff
@@ -56,7 +56,21 @@ def add_to_shed_tool_config(self, shed_tool_conf_dict, elem_list):
                 config_elems.append(elem)
             # Add the new elements to the in-memory list of config_elems.
             for elem_entry in elem_list:
-                config_elems.append(elem_entry)
+                section_found = False
+                if elem_entry.tag == 'section':
+                    # See if there is already a section with a matching ID
+                    for existing_elem in config_elems:
+                        if existing_elem.tag == 'section' and existing_elem.attrib.get('id', None) == elem_entry.attrib.get('id', None):
+                            section_found = True
+                            for child in elem_entry:
+                                existing_elem.append(child)
+                                break
+                            if section_found:
```

Comment:
`section_found` is here always true, left?

Reply:
It's set to `False` for each iteration through `elem_list`

Reply:
At line 64 you're setting this to `True`. I don't see a condition where this could be False after that. The code is not wrong, it's just confusing, you can just break here unconditionally.

Reply:
@mvdbeek that's a good point about the unconditional break, I removed the condition and added some more comments to clarify the flow.